addons/html_editor/static/src/others/x2many_image_field.js

```diff
@@ -0,0 +1,69 @@
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+import { ImageField, imageField } from "@web/views/fields/image/image_field";
+import { X2ManyMediaDialog } from "./x2many_media_viewer";
+import { getVideoUrl } from "@html_editor/utils/url";
+
+export class X2ManyImageField extends ImageField {
+    static template = "html_editor.ImageField";
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        this.dialog = useService("dialog");
+    }
+
+    /**
+     * New method and a new edit button is introduced here to overwrite,
+     * standard behavior of opening file input box in order to update a record.
+     */
+    onFileEdit(ev) {
+        const isVideo = this.props.record.data.video_url;
+        let mediaEl;
+        if (isVideo) {
+            mediaEl = document.createElement("img");
+            mediaEl.dataset.src = this.props.record.data.video_url;
+        }
+        this.dialog.add(X2ManyMediaDialog, {
```

Comment:
Since you moved `X2ManyMediaDialog` into a separate file and renamed it to `CustomMediaDialog`, you forgot to update it's reference here.